### PR TITLE
Update kotlin version in examples

### PR DESCRIPTION
**Pull request by eymar**
Changes:
- update all examples to Kotlin 2.0
- migrate to using kotlin("plugin.compose") for Compose compiler configuration
- remove deprecated Compose Web gradle API usages

- added a workaround for https://github.com/JetBrains/compose-multiplatform/issues/4848 in Jetsnack example

Testing:
- `./gradlew build` in every example
- run some examples in simulators (or desktop, browser)

### CODE REVIEW

File: examples/imageviewer/gradle.properties [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-# TODO: enable configuration cache once we update on kotlin 2.0
-# see https://youtrack.jetbrains.com/issue/KT-64851 (k/wasm issue)
-org.gradle.configuration-cache=false
+org.gradle.configuration-cache=true
 org.gradle.caching=true
 org.jetbrains.compose.experimental.jscanvas.enabled=true
 org.jetbrains.compose.experimental.macos.enabled=true
 org.jetbrains.compose.experimental.wasm.enabled=true
-kotlin.version=1.9.24
+kotlin.version=2.0.0-RC2
```
Comment (MatkovIvan) [inline examples/imageviewer/gradle.properties:11]: Why not `2.0.0`?

Comment (eymar) [reply examples/imageviewer/gradle.properties:11]: Typo. I tried to see if the issue with k/wasm present on RC. 
There is a TODO in the description. So the PR is not complete yet

**Review by igordmn — APPROVED** [review]

File: examples/jetsnack/gradle.properties [hunk 2]
```diff
@@ -8,6 +8,3 @@ compose.version=1.6.10
 # TODO: enable configuration cache once we update on kotlin 2.0
 # see https://youtrack.jetbrains.com/issue/KT-64851 (k/wasm issue)
 org.gradle.configuration-cache=false
```
Comment (MatkovIvan) [inline examples/jetsnack/gradle.properties:10]: It looks like it should be already solved

Comment (eymar) [reply examples/jetsnack/gradle.properties]: fixed

**Pull request merged**